File: ios-cownit/CownItIOS/Models/AppModels.swift
//
//  AppModels.swift
//  CownItIOS
//
//  Roles and role-aware navigation model.
//

import SwiftUI

/// The three workspace roles available in the top-right switcher.
enum Role: String, CaseIterable, Identifiable {
    case individual
    case hr
    case ceo

    var id: String { rawValue }

    var label: String {
        switch self {
        case .individual: return "Individual"
        case .hr: return "HR"
        case .ceo: return "CEO"
        }
    }
}

/// A single entry in the role-aware sidebar.
struct NavItem: Identifiable, Equatable {
    let id: String
    let label: String
    let icon: String

    static func == (lhs: NavItem, rhs: NavItem) -> Bool { lhs.id == rhs.id }
}

extension Role {
    /// Sidebar contents change depending on the active role.
    var navItems: [NavItem] {
        switch self {
        case .individual:
            return [
                NavItem(id: "dashboard", label: "Dashboard", icon: "square.grid.2x2"),
                NavItem(id: "commitments", label: "My Commitments", icon: "checkmark.circle"),
                NavItem(id: "goals", label: "Goals", icon: "flag"),
                NavItem(id: "checkins", label: "Check-ins", icon: "calendar.badge.checkmark"),
                NavItem(id: "profile", label: "Profile", icon: "person"),
            ]
        case .hr:
            return [
                NavItem(id: "overview", label: "Overview", icon: "square.grid.2x2"),
                NavItem(id: "people", label: "People", icon: "person.3"),
                NavItem(id: "commitments", label: "Commitments", icon: "list.bullet.clipboard"),
                NavItem(id: "reviews", label: "Reviews", icon: "star.bubble"),
                NavItem(id: "reports", label: "Reports", icon: "chart.bar"),
            ]
        case .ceo:
            return [
                NavItem(id: "company", label: "Company", icon: "building.2"),
                NavItem(id: "departments", label: "Departments", icon: "point.3.connected.trianglepath.dotted"),
                NavItem(id: "performance", label: "Performance", icon: "chart.line.uptrend.xyaxis"),
                NavItem(id: "alignment", label: "Alignment", icon: "target"),
                NavItem(id: "insights", label: "Insights", icon: "sparkles"),
            ]
        }
    }
}

File: android/app/src/main/java/com/rork/cownit/ui/model/AppModels.kt
package com.rork.cownit.ui.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.AccountTree
import androidx.compose.material.icons.outlined.Assignment
import androidx.compose.material.icons.outlined.BarChart
import androidx.compose.material.icons.outlined.Business
import androidx.compose.material.icons.outlined.CheckCircle
import androidx.compose.material.icons.outlined.Dashboard
import androidx.compose.material.icons.outlined.EventAvailable
import androidx.compose.material.icons.outlined.Flag
import androidx.compose.material.icons.outlined.Groups
import androidx.compose.material.icons.outlined.Insights
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material.icons.outlined.RateReview
import androidx.compose.material.icons.outlined.TrackChanges
import androidx.compose.material.icons.outlined.TrendingUp
import androidx.compose.ui.graphics.vector.ImageVector

/** The three workspace roles available in the top-right switcher. */
enum class Role(val label: String) {
    INDIVIDUAL("Individual"),
    HR("HR"),
    CEO("CEO"),
}

/** A single entry in the role-aware sidebar. */
data class NavItem(
    val id: String,
    val label: String,
    val icon: ImageVector,
)

/** Sidebar contents change depending on the active role. */
fun navItemsFor(role: Role): List<NavItem> = when (role) {
    Role.INDIVIDUAL -> listOf(
        NavItem("dashboard", "Dashboard", Icons.Outlined.Dashboard),
        NavItem("commitments", "My Commitments", Icons.Outlined.CheckCircle),
        NavItem("goals", "Goals", Icons.Outlined.Flag),
        NavItem("checkins", "Check-ins", Icons.Outlined.EventAvailable),
        NavItem("profile", "Profile", Icons.Outlined.Person),
    )
    Role.HR -> listOf(
        NavItem("overview", "Overview", Icons.Outlined.Dashboard),
        NavItem("people", "People", Icons.Outlined.Groups),
        NavItem("commitments", "Commitments", Icons.Outlined.Assignment),
        NavItem("reviews", "Reviews", Icons.Outlined.RateReview),
        NavItem("reports", "Reports", Icons.Outlined.BarChart),
    )
    Role.CEO -> listOf(
        NavItem("company", "Company", Icons.Outlined.Business),
        NavItem("departments", "Departments", Icons.Outlined.AccountTree),
        NavItem("performance", "Performance", Icons.Outlined.TrendingUp),
        NavItem("alignment", "Alignment", Icons.Outlined.TrackChanges),
        NavItem("insights", "Insights", Icons.Outlined.Insights),
    )
}
